src/Microsoft.Diagnostics.Monitoring.RestServer/OutputStreamResult.cs

```diff
@@ -16,17 +19,25 @@ internal sealed class OutputStreamResult : ActionResult
         private readonly Func<Stream, CancellationToken, Task> _action;
         private readonly string _contentType;
         private readonly string _fileDownloadName;
+        private readonly KeyValueLogScope _scope;
 
-        public OutputStreamResult(Func<Stream, CancellationToken, Task> action, string contentType, string fileDownloadName = null)
+        public OutputStreamResult(Func<Stream, CancellationToken, Task> action, string contentType, string fileDownloadName, KeyValueLogScope scope)
```

Comment:
nit: Consider passing in the specific information you'll want to push into the scope stack, rather than KeyValueLogScope directly. I don't think we should tie up class apis with implementation detail logger structures.

Reply:
To do what you suggested, I would have to provide the IProcessInfo and the artifact type as parameters to the result objects, which would only be used for logging scopes; if we add any other logging scope information in the future, it too would need to be added as additional parameters.

I felt the way I have it here separates the additional logging scope information concern from what the result objects actually care about.